relay-general/src/pii/processor.rs

```diff
@@ -317,42 +317,49 @@ impl<'a> Processor for PiiProcessor<'a> {
         _meta: &mut Meta,
         state: &ProcessingState,
     ) -> ProcessingResult {
-        // View pairlists as objects just for the purpose of PII stripping (e.g. `event.tags.mykey`
-        // instead of `event.tags.42.0`). For other purposes such as trimming we would run into
-        // problems:
-        //
-        // * tag keys need to be trimmed too and therefore need to have a path
-
-        for (idx, annotated) in value.iter_mut().enumerate() {
-            if let Some(ref mut pair) = annotated.value_mut() {
-                let (ref mut key, ref mut value) = pair.as_pair_mut();
-                if let Some(ref key_name) = key.as_str() {
-                    // if the pair has no key name, we skip over it for PII stripping. It is
-                    // still processed with index-based path in the invocation of
-                    // `process_child_values`.
-                    process_value(
-                        value,
-                        self,
-                        &state.enter_borrowed(
-                            key_name,
-                            state.inner_attrs(),
-                            ValueType::for_field(value),
-                        ),
-                    )?;
-                } else {
-                    process_value(
-                        value,
-                        self,
-                        &state.enter_index(idx, state.inner_attrs(), ValueType::for_field(value)),
-                    )?;
-                }
+        process_pairlist(self, value, state)
+    }
+}
+
+pub fn process_pairlist<P: Processor, T: ProcessValue + AsPair>(
```

Comment:
_nit_: Could you add a doc comment for this function.

I'm wondering if there's a better place for this than as a top-level function in this module. WDYT?

Reply:
I moved it out, didn't want to start a new file at first :)